servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/KeepAlivePolicies.java

```diff
@@ -0,0 +1,143 @@
+ */
+public final class KeepAlivePolicies {
+    static final KeepAlivePolicy DISABLE_KEEP_ALIVE =
+            new DefaultKeepAlivePolicy(ofDays(365), ofDays(365), false);
+
+    private KeepAlivePolicies() {
+        // no instances.
+    }
+
+    /**
+     * Returns a {@link KeepAlivePolicy} that disables all keep alive behaviors.
+     *
+     * @return A {@link KeepAlivePolicy} that disables all keep alive behaviors.
+     */
+    public static KeepAlivePolicy disabled() {
+        return DISABLE_KEEP_ALIVE;
+    }
+
+    /**
+     * Returns a {@link KeepAlivePolicy} that sends a <a href="https://tools.ietf.org/html/rfc7540#section-6.7">
+     * ping</a> if the channel is idle for the passed {@code idleDuration}. Default values are used for other parameters
+     * of the returned {@link KeepAlivePolicy}.
+     *
+     * @param idleDuration {@link Duration} of idleness on a connection after which a
+     * <a href="https://tools.ietf.org/html/rfc7540#section-6.7">ping</a> is sent.
+     * @return A {@link KeepAlivePolicy} that disables all keep alive behaviors.
+     */
+    public static KeepAlivePolicy whenIdleFor(final Duration idleDuration) {
+        return new KeepAlivePolicyBuilder().idleDuration(idleDuration).build();
+    }
+
+    /**
+     * Returns a {@link KeepAlivePolicy} that sends a <a href="https://tools.ietf.org/html/rfc7540#section-6.7">
+     * ping</a> if the channel is idle for the passed {@code idleDuration} and waits for {@code ackTimeout} for an ack
+     * for that <a href="https://tools.ietf.org/html/rfc7540#section-6.7">ping</a>. Default values are used for other
+     * parameters of the returned {@link KeepAlivePolicy}.
+     *
+     * @param idleDuration {@link Duration} of idleness on a connection after which a
+     * <a href="https://tools.ietf.org/html/rfc7540#section-6.7">ping</a> is sent.
+     * @param ackTimeout {@link Duration} to wait for an acknowledgment of a previously sent
+     * <a href="https://tools.ietf.org/html/rfc7540#section-6.7">ping</a>.
+     * @return A {@link KeepAlivePolicy} that disables all keep alive behaviors.
+     */
+    public static KeepAlivePolicy whenIdleFor(final Duration idleDuration, final Duration ackTimeout) {
+        return new KeepAlivePolicyBuilder().idleDuration(idleDuration).ackTimeout(ackTimeout).build();
+    }
+
+    /**
+     * A builder of {@link KeepAlivePolicy}.
+     */
+    public static final class KeepAlivePolicyBuilder {
+        private Duration idleDuration = DEFAULT_IDLE_DURATION;
+        private Duration ackTimeout = DEFAULT_ACK_TIMEOUT;
+        private boolean withoutActiveStreams;
+
+        /**
+         * Set the {@link Duration} of idleness on a connection after which a
+         * <a href="https://tools.ietf.org/html/rfc7540#section-6.7">ping</a> is sent.
+         * <p>
+         * <strong>Too short ping durations may cause high network traffic, implementations may enforce a minimum
```

Comment:
Since this is a final class now instead of an interface, "implementation_s_ may enforce" is not correct anymore. Consider something like:
> Too short ping durations may cause high network traffic. Too small duration is not allowed.

Or be specific about 10 seconds.

Reply:
Good point. I will avoid mentioning a value in the doc though.